Skip bandwidth bound mms

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #101915
* #102200
* __->__ #102199
* #101913

Speeds up compilation time, and was particularly needed for cm3leon_generate which has a ton of small matmuls of different sizes.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10

eellison committed:
Skip bandwidth bound mms

[ghstack-poisoned]

eellison committed:
Update on "Skip bandwidth bound mms"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10

[ghstack-poisoned]

eellison committed:
Update on "Skip bandwidth bound mms"


Speeds up compilation time, and was particularly needed for cm3leon_generate which has a ton of small matmuls of different sizes.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10

[ghstack-poisoned]

ngimel commented on torch/_inductor/fx_passes/pad_mm.py:
```diff
@@ -149,6 +149,29 @@ def pad_addmm(
+
+    machine_balance = (1000 * get_flops(dtype)) / get_dram_gbps()
+
+    return arithmetic_intensity > machine_balance
```
do you need some coefficient here, for safety?

eellison committed:
Update on "Skip bandwidth bound mms"


Speeds up compilation time, and was particularly needed for cm3leon_generate which has a ton of small matmuls of different sizes.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10

[ghstack-poisoned]

eellison commented:
@pytorchbot merge

eellison commented:
@pytorchbot merge
